api.ts

```diff
@@ -957,15 +959,87 @@ export function isSyncPackFormula(fn: BaseFormula<ParamDefs, any>): fn is Generi
  * that the sync formula should be invoked again to get a next page of results. Sync functions
  * are called repeatedly until there is no continuation returned.
  */
-export interface SyncFormulaResult<K extends string, L extends string, SchemaT extends ObjectSchemaDefinition<K, L>> {
-  /** The list of results from this page. */
+export type SyncFormulaResult<K extends string, L extends string, SchemaT extends ObjectSchemaDefinition<K, L>> =
+  | SyncFormulaPartialResult<K, L, SchemaT>
+  | SyncFormulaFinalResult<K, L, SchemaT>;
+
+/** @hidden */
+interface BaseSyncFormulaResult<K extends string, L extends string, SchemaT extends ObjectSchemaDefinition<K, L>> {
+  /** The list of rows from this page. */
   result: Array<ObjectSchemaDefinitionType<K, L, SchemaT>>;
+}
+
+/**
+ * A result from a sync table formula invocation that signals that there are more results to fetch for the
+ * current execution.
+ */
+export interface SyncFormulaPartialResult<
+  K extends string,
+  L extends string,
+  SchemaT extends ObjectSchemaDefinition<K, L>,
+> extends BaseSyncFormulaResult<K, L, SchemaT> {
   /**
    * A marker indicating where the next sync formula invocation should pick up to get the next page of results.
    * The contents of this object are entirely of your choosing. Sync formulas are called repeatedly
    * until there is no continuation returned.
    */
-  continuation?: Continuation;
+  continuation: Continuation;
+}
+
+/**
+ * A result from a sync table formula invocation that signals that there are no more results to fetch for the
+ * current execution.
+ */
+export interface SyncFormulaFinalResult<
+  K extends string,
+  L extends string,
+  SchemaT extends ObjectSchemaDefinition<K, L>,
+> extends BaseSyncFormulaResult<K, L, SchemaT> {
+  /**
+   * Once there is no additional continuation returned from a pack sync formula, this may be returned instead, to give
+   * metadata about the entirety of the sync execution.
+   *
+   * TODO(patrick): Unhide this
+   * @hidden
+   */
+  syncExecutionCompletionMetadata?: Omit<
```

Comment:
Nit, I would probably give this type a name. It'll eventually show up in documentation too and it'll be clearer to be a named thing.